### DISCUSSION TITLE
Github Copilot 寫程式的最佳夥伴

### DISCUSSION THREAD

**Discussion by KH1126** [1101-GS4719-C/Python-Community-GS4719-C#1918] [General]
最近申請到Github Copilot，在實際運用過後，
可以發現寫程式的效率大幅提升。
在vscode的擴充功能安裝github copilot，通過授權後即可使用。
(github copilot有學生方案，通過github education可申請到免費使用github copilot的權利)

![image-6](https://github.com/NCU-GS4719-Python/Python-Community/assets/100942642/c43054c1-22dc-4868-a311-fc56e1115eba)

此連結為我找到簡易介紹github copilot的文章，
[https://useme.medium.com/%E8%83%BD%E7%AF%80%E7%9C%81%E5%9B%9B%E6%88%90%E9%96%8B%E7%99%BC%E6%99%82%E9%96%93%E7%9A%84%E5%B7%A5%E5%85%B7-github-copilot-%E7%9A%84%E4%BB%8B%E7%B4%B9%E8%88%87%E5%BF%83%E5%BE%97%E5%88%86%E4%BA%AB-2a07e8ce8f6d](https://useme.medium.com/%E8%83%BD%E7%AF%80%E7%9C%81%E5%9B%9B%E6%88%90%E9%96%8B%E7%99%BC%E6%99%82%E9%96%93%E7%9A%84%E5%B7%A5%E5%85%B7-github-copilot-%E7%9A%84%E4%BB%8B%E7%B4%B9%E8%88%87%E5%BF%83%E5%BE%97%E5%88%86%E4%BA%AB-2a07e8ce8f6d)

通過文章可了解到在有github copilot的輔助下，開發者可減少約四成的程式碼，
有此可見github copilot快捷指令的便利性。
如果使用得當，甚至有可能改變開發人員的編碼方式。

我認為雖然 Github Copilot 能夠顯著提升開發效率，但合理運用它同樣至關重要。
人工智能是一種工具，工程師可以透過使用自動化來簡化工作流程，但同時也應該警惕其產生的輸出是否符合需求。
因此在使用 Copilot 時，我願意將瑣碎的任務交給它，但與此同時，也須警惕它生成的程式碼是否合適並符合品質標準，
以確保其符合項目需求。

**Comment by William08290829** [comment]
額外補充
申請的時候他會有需要`雙重要素驗證 (2FA)`
要先下載`google authenticator`或是其他的身分驗證軟體
如果一次不弄好到位的話 上傳在學證明的照片送出之後 他會馬上拒絕
然後因為它會檢測是不是放上與上次相同照片 導致你又要重弄一張....

**Comment by benbebobo149** [comment]
我覺得你對於Copilot提出的建議非常實用，這讓我想起了在實習或公司中使用新工具時需要注意的事項。在使用Copilot之前，最好先徵求主管的意見，以確保不會觸碰到任何內部機密消息，避免不小心將敏感信息上傳到公開的GitHub上。即使關掉了上傳到雲端的功能，仍然可能存在資安風險。此外，他人在審查你的程式碼時可能難以理解你的思路，這可能會導致沟通和合作上的困難。最後，感謝你分享了這些寶貴的建議，這對我們在工作中更加謹慎和細心地使用工具提供了重要的提醒。

**Comment by benbebobo149** [comment]
我認為你提到的這個問題非常值得關注，尤其是對於個人開發者來說。使用Copilot提供的程式碼建議時，我們可能會忽略程式碼的授權訊息，這可能導致產生版權上的問題，特別是如果使用了特定的授權協議，可能會觸發GPL等開源協議的規定，進而影響到我們的開發項目。雖然目前大家對於這個問題可能關注不夠，但我認為這是一個需要我們密切關注並思考的問題，也許在未來GitHub會提供更多指引來幫助我們應對這個問題。最後，感謝你分享了這個重要的觀點，這讓我們能夠更加警覺和審慎地使用這些開發工具。

**Comment by yyujimmy** [comment]
這篇文章確實提醒了我們在使用新技術時需要保持警惕。雖然Github Copilot等工具能夠顯著提高開發效率，但我認為開發者應該保持對程式碼的主動控制權。我認為Copilot可以作為一個有用的輔助工具，但不應該完全依賴它生成的程式碼。

作為開發者，我們應該保持對項目的全面了解，包括需求、架構和品質標準。在這種情況下，我認為最好的方法是將Copilot用於執行瑣碎的任務和重複性的代碼，同時對於核心邏輯和重要功能則應該由人工進行編碼。這樣可以最大程度地利用Copilot的優勢，同時保持程式碼的一致性和品質。

總之，我認為使用Github Copilot等工具需要找到一個平衡點，既能夠提高效率，又能夠保持程式碼的品質和適用性。這需要開發者在使用這些工具時保持警惕和謹慎，並始終將最終產品的品質放在首位。

**Comment by wanderer657** [comment]
最近用了Github Copilot，寫程式快到爆！安裝了vscode的擴充，輕鬆授權就能用。跟其他工具比，我認為Copilot可以省下四分之一的碼，一點也不浮誇，但得小心點，它雖然能提速，但得用得準，不能光憑它生成的碼，得確保品質，要不然爛出天際。

**Comment by hbh10601** [comment]
這工具在我去參加微軟的工作坊課程時，那邊的人也有介紹給我們使用，那時候嘗試了一下之後，讓我覺得之前很多邏輯part花時間手打程式碼根本在浪費時間，這東西超好用。
不過使用者也要有基本的程式邏輯才知道怎麼向他尋求幫助。

**Comment by sunset513** [comment]
GitHub Copilot真的十分好用，它大大的提升我的開發效率。

有時候我會在腦內構思一個功能，之後再將想法寫成註解。copilot 能夠聰明的根據註解，就將這個功能寫出來。我用過了一段時間之後，就深深的愛上他❤️回不去了。

我無法想像沒有他的日子，我會受到多大的影響，這會是多麼的難受。
